model/trigger/build.go

```diff
@@ -208,3 +82,135 @@ func taskStatusSubformat(n int, verb string) string {
 func appendTime(b *build.Build, txt string) string {
 	return fmt.Sprintf("%s in %s", txt, b.FinishTime.Sub(b.StartTime).String())
 }
+
+type buildTriggers struct {
+	event    *event.EventLogEntry
+	data     *event.BuildEventData
+	build    *build.Build
+	uiConfig evergreen.UIConfig
```

Comment:
I think the UI config should be moved out of the interface implementations and be handled in makeCommonPayload, since it's all the same and conceptually isn't a part of the trigger really

Reply:
There's no need to have it be part of the implementation though since it's always the same and isn't related to the trigger itself

Reply:
I understand completely, it seems silly and redundant. But I don't want makeCommonPayload to make the payload AND fetch data just because it's commonly used. I want the side effect of data fetching somewhere else, Fetch being the most appropriate place. Perhaps moving it into base is an option, but I definitely don't want to move it into makeCommonPayload.